pkgs/build-support/fetchtorrent/tests.nix

```diff
@@ -1,47 +1,58 @@
+
+  # All routes to download the torrent should produce the same output and
+  # therefore have the same FOD hash.
+  hash = "sha256-EzbmBiTEWOlFUNaV5R4eDeD9EBbp6d93rfby88ACg0s=";
 in
 
 {
   http-link = testers.invalidateFetcherByDrvHash fetchtorrent {
-    url = "https://webtorrent.io/torrents/wired-cd.torrent";
-    hash = "sha256-OCsC22WuanqoN6lPv5wDT5ZxPcEHDpZ1EgXGvz1SDYo=";
+    inherit hash;
+    url = httpUrl;
     backend = "transmission";
-    inherit (wired-cd) meta;
+    inherit (sintel) meta;
   };
   magnet-link = testers.invalidateFetcherByDrvHash fetchtorrent {
-    url = "magnet:?xt=urn:btih:a88fda5954e89178c372716a6a78b8180ed4dad3&dn=The+WIRED+CD+-+Rip.+Sample.+Mash.+Share&tr=udp%3A%2F%2Fexplodie.org%3A6969&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969&tr=udp%3A%2F%2Ftracker.empire-js.us%3A1337&tr=udp%3A%2F%2Ftracker.leechers-paradise.org%3A6969&tr=udp%3A%2F%2Ftracker.opentrackr.org%3A1337&tr=wss%3A%2F%2Ftracker.btorrent.xyz&tr=wss%3A%2F%2Ftracker.fastcast.nz&tr=wss%3A%2F%2Ftracker.openwebtorrent.com&ws=https%3A%2F%2Fwebtorrent.io%2Ftorrents%2F&xs=https%3A%2F%2Fwebtorrent.io%2Ftorrents%2Fwired-cd.torrent";
-    hash = "sha256-OCsC22WuanqoN6lPv5wDT5ZxPcEHDpZ1EgXGvz1SDYo=";
+    inherit hash;
+    url = magnetUrl;
     backend = "transmission";
-    inherit (wired-cd) meta;
+    inherit (sintel) meta;
   };
   http-link-rqbit = testers.invalidateFetcherByDrvHash fetchtorrent {
-    url = "https://webtorrent.io/torrents/wired-cd.torrent";
-    hash = "sha256-OCsC22WuanqoN6lPv5wDT5ZxPcEHDpZ1EgXGvz1SDYo=";
+    inherit hash;
+    url = httpUrl;
     backend = "rqbit";
-    inherit (wired-cd) meta;
+    meta = sintel.meta // {
+      broken = true;
+    };
   };
   magnet-link-rqbit = testers.invalidateFetcherByDrvHash fetchtorrent {
-    url = "magnet:?xt=urn:btih:a88fda5954e89178c372716a6a78b8180ed4dad3&dn=The+WIRED+CD+-+Rip.+Sample.+Mash.+Share&tr=udp%3A%2F%2Fexplodie.org%3A6969&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969&tr=udp%3A%2F%2Ftracker.empire-js.us%3A1337&tr=udp%3A%2F%2Ftracker.leechers-paradise.org%3A6969&tr=udp%3A%2F%2Ftracker.opentrackr.org%3A1337&tr=wss%3A%2F%2Ftracker.btorrent.xyz&tr=wss%3A%2F%2Ftracker.fastcast.nz&tr=wss%3A%2F%2Ftracker.openwebtorrent.com&ws=https%3A%2F%2Fwebtorrent.io%2Ftorrents%2F&xs=https%3A%2F%2Fwebtorrent.io%2Ftorrents%2Fwired-cd.torrent";
-    hash = "sha256-OCsC22WuanqoN6lPv5wDT5ZxPcEHDpZ1EgXGvz1SDYo=";
+    inherit hash;
+    url = magnetUrl;
     backend = "rqbit";
-    inherit (wired-cd) meta;
+    meta = sintel.meta // {
+      broken = true;
```

Comment:
This should probably be done in the fetcher instead; otherwise callers will not notice the problem.
Alternatively, after reading #432001, it seems that it may be an easy fix?

Reply:
I thought about that, but decided against it because the fetcher isn't broken, it just has different behaviour depending if it's using certain versions of rqbit.  I have no way of telling which behaviour people consider to be correct at present (and it could well be that different people have good reason to believe both behaviours to be correct).

Regardless, the test _is_ currently broken, because it's expecting behaviour that the fetcher doesn't currently provide.